core/src/autogluon/core/ray/resources_calculator.py

```diff
@@ -19,25 +19,39 @@ def calc_type(self):
         raise NotImplementedError
 
     @staticmethod
-    def get_total_gpu_count(user_specified_num_gpus: int, model_default_num_gpus: int, num_gpus_available: Optional[float] = None):
+    def get_total_gpu_count(
+        model_default_num_gpus: Union[int, float],
+        user_specified_num_gpus: Optional[Union[int, float]] = None
+    ):
+        """
+        model_minimum_num_gpus
+            Number of gpus needed by the model minimum
+        user_specified_num_gpus
+            Number of gpus specified by the user.
+            This corresponds to the number of gpus used by the hpo experiment(shared by all trials when training a model)
+            or the bagging of the model(shared by all bags)
+        """
         if user_specified_num_gpus is not None:
-            total_gpu = get_gpu_count_all() if num_gpus_available is None else min(get_gpu_count_all(), num_gpus_available)
-            num_gpus = min(user_specified_num_gpus, total_gpu)
+            num_gpus = min(user_specified_num_gpus, get_gpu_count_all())
+            if num_gpus < user_specified_num_gpus:
+                logger.warning(f'Specified num_gpus = {user_specified_num_gpus}, but only {num_gpus} are available')
```

Comment:
Should this only be a warning? Would it make sense to raise an exception instead?